src/main/java/net/syscon/elite/repository/jpa/model/OffenderNonAssociationDetail.java

```diff
@@ -0,0 +1,110 @@
+package net.syscon.elite.repository.jpa.model;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+import org.hibernate.annotations.JoinColumnOrFormula;
+import org.hibernate.annotations.JoinColumnsOrFormulas;
+import org.hibernate.annotations.JoinFormula;
+import org.hibernate.annotations.NotFound;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.IdClass;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+import java.io.Serializable;
+import java.time.LocalDateTime;
+
+import static org.hibernate.annotations.NotFoundAction.IGNORE;
+
+@Getter
+@Entity
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@EqualsAndHashCode(callSuper = false)
+@ToString(exclude = {"offender", "nsOffender", "offenderBooking"})
+@Table(name = "OFFENDER_NA_DETAILS")
+@IdClass(OffenderNonAssociationDetail.Pk.class)
+public class OffenderNonAssociationDetail extends AuditableEntity {
+
+    @AllArgsConstructor
+    @NoArgsConstructor
+    @Getter
+    public static class Pk implements Serializable {
+        private Offender offender;
+
+        private Offender nsOffender;
+
+        private Integer typeSequence;
+    }
+
+    @Id
+    @ManyToOne(optional = false)
+    @JoinColumn(name = "OFFENDER_ID", nullable = false, insertable = false, updatable = false)
+    private Offender offender;
+
+    @Id
+    @ManyToOne(optional = false)
+    @JoinColumn(name = "NS_OFFENDER_ID", nullable = false, insertable = false, updatable = false, referencedColumnName = "OFFENDER_ID")
+    private Offender nsOffender;
+
+    @Id
+    @Column(name = "TYPE_SEQ", nullable = false, insertable = false, updatable = false)
+    private Integer typeSequence;
```

Comment:
Is it safe to use Integer here over Long?

Reply:
Schema is a number(3) so yes.